### ISSUE TITLE
Matrices in parentheses not rendering correctly

### ISSUE THREAD

**Issue by jamiemlaw** [christianp/asciimath2tex#26]
I'm using asciimath2tex@1.4.0. There seems to be something funky going on if I wrap a matrix in parentheses. I noticed this when trying to mark up a matrix determinant with the following input:

```js
parser.parse('det([[a,b],[c,d]])');
```

Expected output:
```js
'\det{\left ( \left [ \begin{matrix} a & b \\ c & d \end{matrix} \right ] \right )}'
```

Actual output:
```js
'\det{\left ( \begin{matrix} \left [ a , b \right ] & \left [ c , d \right ] \end{matrix} \right )}'
```

It seems that the minimal test case that fails is

```js
parser.parse('([[a],[b]])');
```

Expected output:
```js
'\left ( \left [ \begin{matrix} a \\ b \end{matrix} \right ] \right )'
```

Actual output:
```js
'\left ( \begin{matrix} \left [ a \right ] & \left [ b \right ] \end{matrix} \right )'
```

**Comment by jamiemlaw** [comment]
Found a workaround:

```js
parser.parse('([{:[a],[b]:}])');
```

which produces

```js
'\left ( \left [ \left . \begin{matrix} a \\ b \end{matrix} \right . \right ] \right )'
```

**Comment by christianp** [comment]
I've had this issue in my inbox for exactly a year, and today I came up with a solution! 
The root of the problem was that my parser works in a completely different order to the original AsciiMathML, so I couldn't just copy their logic for spotting matrices.
I realised that you want to identify a matrix as deep into nested brackets as possible, so I added a part to the grammar to look for a matrix wrapped in extra brackets before just looking for a matrix, allowing it to consume the outer brackets first.
